### PR TITLE
Fix less fallback

### PR DESCRIPTION
**Pull request by l-vincent-l**
There were several mistakes made.
We were comparing arrival time with departure_time in computation of lost_time.
Also, we were computing departure walking time with the destination journey pattern point...

### CODE REVIEW

File: source/routing/raptor_solutions.cpp [hunk 1]
```diff
@@ -151,9 +151,11 @@ get_pareto_front(bool clockwise, const std::vector<std::pair<type::idx_t, naviti
             for(auto spid_dep : departs) {
                 if(raptor.data.pt_data->journey_pattern_points[final_jpp_idx]->stop_point->idx == spid_dep.first) {
                     if(clockwise) {
-                        s.upper_bound = s.upper_bound + (spid_dep.second.total_seconds());
+                        s.upper_bound = s.upper_bound + spid_dep.second.total_seconds();
+                        s.total_arrival = raptor.labels[round][best_jpp].dt_pt - spid_dep.second.total_seconds();
                     }else {
                         s.upper_bound = s.upper_bound - (spid_dep.second.total_seconds());
```
Comment (TeXitoi) [inline source/routing/raptor_solutions.cpp]: you can also remote parenthesis here.

**Comment by TeXitoi** [comment]
Is https://github.com/CanalTP/navitia/blob/dev/source/routing/raptor.cpp#L247 still needed with this patch? (just return true in the lambda and check the unit tests to check it)

**Comment by l-vincent-l** [comment]
why is it related?

File: source/routing/raptor_solutions.cpp [hunk 2]
```diff
@@ -296,7 +305,7 @@ navitia::time_duration getWalkingTime(int count, type::idx_t jpp_idx, const std:
                      bool clockwise, bool disruption_active, const type::AccessibiliteParams & accessibilite_params,
                      const RAPTOR& raptor) {
 
-    const type::JourneyPatternPoint* current_jpp = raptor.data.pt_data->journey_pattern_points[jpp_idx];
+    const auto* current_jpp = raptor.data.pt_data->journey_pattern_points[jpp_idx];
```
Comment (antoine-de) [inline source/routing/raptor_solutions.cpp:308]: the '*' is not mandatory

Comment (TeXitoi) [inline source/routing/raptor_solutions.cpp:308]: it's not true: `const auto` wll give `type::JourneyPatternPoint * const`, and `const auto *` will give `type::JourneyPatternPoint const *`

**Comment by l-vincent-l** [comment]
So this https://github.com/CanalTP/navitia/blob/dev/source/routing/raptor.cpp#L247 is still needed.

**Pull request merged**